File: app/services/mock_data.py
"""
JINNI Grid - Mock Data Service
Portfolio and equity curve only. Fleet data is real (worker_registry). 
app/services/mock_data.py
"""
import random
from datetime import datetime, timedelta, timezone

def get_portfolio_summary() -> dict:
    return {
        "total_balance": 248750.00, "total_equity": 251320.45,
        "floating_pnl": 2570.45, "daily_pnl": 1847.30,
        "open_positions": 12, "realized_pnl": 18432.60,
        "margin_usage": 34.7, "win_rate": 68.5,
    }

def get_equity_history() -> list:
    rng = random.Random(42)
    points = []
    start = datetime(2026, 2, 5, tzinfo=timezone.utc)
    val = 200000.0
    for i in range(90):
        d = start + timedelta(days=i)
        val += (rng.random() - 0.42) * 2000
        if val < 180000: val = 180000.0
        points.append({"timestamp": d.strftime("%Y-%m-%d"), "equity": round(val, 2)})
    return points

File: ui/js/mockData.js
/* mockData.js */

var MockData = (function () {
  'use strict';

  var portfolioStats = {
    totalBalance: 248750.00,
    totalEquity: 251320.45,
    floatingPnl: 2570.45,
    dailyPnl: 1847.30,
    openPositions: 12,
    realizedPnl: 18432.60,
    marginUsage: 34.7,
    winRate: 68.5
  };

  var equityCurve = (function () {
    var points = [];
    var startDate = new Date(2026, 1, 5);
    var value = 200000;
    // Seeded-style random using simple LCG
    var seed = 42;
    function seededRandom() {
      seed = (seed * 16807 + 0) % 2147483647;
      return (seed - 1) / 2147483646;
    }
    for (var i = 0; i < 90; i++) {
      var date = new Date(startDate);
      date.setDate(date.getDate() + i);
      value += (seededRandom() - 0.42) * 2000;
      if (value < 180000) value = 180000;
      points.push({
        date: date.toISOString().slice(0, 10),
        value: Math.round(value * 100) / 100
      });
    }
    return points;
  })();

  return {
    portfolioStats: portfolioStats,
    equityCurve: equityCurve
  };
})();
